src/game.ts

```diff
@@ -1201,9 +1201,13 @@ export default class Game {
 	triggerTrap(trigger, arg) {
 		const [triggeredCreature] = arg;
 
-		triggeredCreature.hexagons.forEach((hex) => {
-			hex.activateTrap(this.triggers[trigger], triggeredCreature);
-		});
+		const traps = getPointFacade().getTrapsAt(triggeredCreature);
```

Comment:
Maybe @DreadKnight can chime in here.

From my point of view:

This will destroy all `Trap`s under a `Creature` as soon as one `Trap` is activated. I don't think that's desired for all types of `Trap` – "Poisonous Vine" is a bit of an outlier.

For now, I think this bug should probably be addressed locally in `src/abilities/Impaler.js` in the "Poisonous Vine" ability.

Reply:
Yeah, good find from @andretchen0, he's right.

Reply:
@andretchen0 I did some testing, with red player's Impaler having upgraded Poisonous Vine on two units and one eventually destroyed the trap, only the trap under it got destroyed, not for the other unit as well. Might do more testing though. Found a bug already, but not directly related to this.

Reply:
I think it should be addressed in `src/abilities/Impaler.js` as well until we get that higher level fix.

Reply:
I think it should be addressed in `src/abilities/Impaler.js` as well until we get that higher level fix.